model/generate.go

```diff
@@ -727,14 +727,24 @@ type projectMaps struct {
 }
 
 // validateMaxTasksAndVariants validates that the GeneratedProject contains fewer than 100 variants and 1000 tasks.
-func (g *GeneratedProject) validateMaxTasksAndVariants() error {
+func (g *GeneratedProject) validateMaxTasksAndVariants(ctx context.Context, p *Project) error {
 	catcher := grip.NewBasicCatcher()
+	bvTasks := p.FindAllBuildVariantTasks()
+	settings, err := evergreen.GetConfig(ctx)
+	if err != nil {
+		return errors.Wrap(err, "getting admin settings")
+	}
 	if len(g.BuildVariants) > maxGeneratedBuildVariants {
 		catcher.Errorf("it is illegal to generate more than %d buildvariants", maxGeneratedBuildVariants)
 	}
 	if len(g.Tasks) > maxGeneratedTasks {
 		catcher.Errorf("it is illegal to generate more than %d tasks", maxGeneratedTasks)
 	}
+	totalTasks := len(bvTasks) + len(g.Tasks)
```

Comment:
Is there a reason we count both tasks in the BV list + task definitions as a task in a version? I would have thought that the BV task list would be the thing being limited since max tasks per version is measuring how many tasks could run, not how many are defined. The number of task definitions per generate.tasks is already capped at 25k from the check above this.

Reply:
Here the bv tasks list come from the version before it generated any tasks, so my reasoning was summing this with the number of task definitions would result in the new number of tasks within the version. However it looks like the length of the task definitions isn't the same as the number of tasks to be generated by the command, so I moved this validation check elsewhere.